### DISCUSSION TITLE
MQTT support

### DISCUSSION THREAD

**Discussion by simplayy** [openthread/openthread#8733] [Q&A] [answered]
trying to follow the openthread guides I managed to create the network and send packets to a coap server.  it would be useful to do the same thing but with MQTT protocol but I haven't found the API for MQTT on openthread

**Comment by jwhui** [comment, marked as answer]
Thread is a network technology for transporting IPv6 datagrams. Thread is intentionally application layer agnostic.

OpenThread does provide an implementation of CoAP simply because Thread requires the use of CoAP for its own control messaging. However, beyond CoAP, OpenThread does not implement any other application-layer protocol, including MQTT, Matter, and others.

Given that MQTT is an IP-based protocol, it should be fairly straight-forward to support MQTT over Thread.

**Comment by fgervais** [comment]
If it helps, I give you the link of one of my project that does just that: https://github.com/fgervais/project-nrf-thread-switch
